src/interface.jl

```diff
@@ -121,57 +229,93 @@ struct LogFuncDensity{F}
 end
 LogFuncDensity
 
-@inline hasdensity(::LogFuncDensity) = true
+@inline densitykind(::LogFuncDensity) = IsDensity
 
-@inline logdensityof(d::LogFuncDensity, x) = d._log_f(x)
-@inline logdensityof(d::LogFuncDensity) = d._log_f
+@inline logdensityof(object::LogFuncDensity, x) = object._log_f(x)
+@inline logdensityof(object::LogFuncDensity) = object._log_f
 
-function Base.show(io::IO, d::LogFuncDensity)
-    print(io, nameof(typeof(d)), "(")
-    show(io, d._log_f)
+@inline densityof(object::LogFuncDensity, x) = exp(object._log_f(x))
+@inline densityof(object::LogFuncDensity) = exp ∘ object._log_f
+
+function Base.show(io::IO, object::LogFuncDensity)
+    print(io, nameof(typeof(object)), "(")
+    show(io, object._log_f)
     print(io, ")")
 end
 
 
 
 """
-    densityof(d, x)::Real
+    funcdensity(f)
```

Comment:
`densityfunction`?

Reply:
`densityfunction` to me, would sound like the result is a function, not a density "object" created *from* a function.